src/abi_x86_64.cpp

```diff
@@ -151,7 +151,7 @@ void classifyType(Classification& accum, jl_datatype_t *dt, uint64_t offset) con
         size_t i;
         for (i = 0; i < jl_datatype_nfields(dt); ++i) {
             jl_value_t *ty = jl_field_type(dt, i);
-            if (!jl_is_datatype(ty) || ((jl_datatype_t*)ty)->layout == NULL || jl_is_array_type(ty))
+            if (jl_field_isptr(dt, i))
```

Comment:
I don't think `->layout` is necessarily defined. I'm unsure the best way to match `_julia_struct_to_llvm` here, as it seems many other abi files might similarly be trying to access it.

Reply:
I updated a couple other cases; see what you think.